### PR TITLE
Feature/persist ile selections

### PR DESCRIPTION
**Pull request by mheiman**
Addresses #7486

This PR adds several enhancements to the ILE toolbar:
- Selected items are stored within the tab session, so a selection (and deselection) can be done across multiple search result pages or other browsing.
- The toolbar now provides feedback when different types are selected in the same session
- The Merge Works button is now visible when both works and editions are selected
- The toolbar is now only visible when in use, and its appearance has been made slightly more prominent to reduce the risk of accidental selection carryover.

### Technical
Selected items are stored in tab-specific storage until manually cleared, in an object containing arrays of olids keyed by type: `{'edition':['OL1M','OL2M'], 'work':[],...}` The `work` array may contain entries of the form `OL3W:OL4M` where the second value is the id of the specific edition surfaced in a search result.

### Testing
Try a variety of selections on search, author and work pages. Set up a merge. Drag works and editions between authors and works.

### Screenshot
Screenshot 2023-02-03 at 3 07 29 PM

### Stakeholders
@cdrini @mek @mheiman

### CODE REVIEW

**Review by mekarpeles — APPROVED** [review]
Other than the failing js test, changes lgtm, though deferring to @cdrini (domain expert) on approving.

Nice work!

**Comment by mheiman** [comment]
There's a SelectionManager.test.js that's now failing since I completely changed how the getSelectedItems() method works. It's now reliant on the browser context, so I can't just update the test to make it work again. What's the best path forward?

File: openlibrary/plugins/openlibrary/js/ile/utils/SelectionManager/SelectionManager.js [hunk 1]
```diff
@@ -37,17 +45,34 @@ export default class SelectionManager {
...
+                        }
+                    }
+                }
             }
         }
 
+        // Populate the status bar images for stored selections
+        for (const type of SelectionManager.TYPES) {
+            this.selectedItems[type.singular].forEach(item => {
+                this.ile.$statusImages.append(`<li><img title="${item}" src="${type.image(item)}" /></li>`);
```
Comment (cdrini) [inline openlibrary/plugins/openlibrary/js/ile/utils/SelectionManager/SelectionManager.js:70]:
```suggestion
            this.selectedItems[type.singular].forEach(olid => {
                this.ile.$statusImages.append(`<li><img title="${olid}" src="${type.image(olid)}" /></li>`);
```

**Review by cdrini — APPROVED** [review]

**Comment by cdrini** [comment]
Tested via gitpod, everything works swell! Super excited to see this go out, it'll make so many things much easier! Thank you @mheiman !!

**Pull request merged**